src/Types/RelationParserHelper.php

```diff
@@ -112,24 +92,91 @@ public function findRelatedModelInRelationMethod(
         return $this->reflectionProvider->hasClass($returnClass) ? $returnClass : null;
     }
 
-    /**
-     * @param  string  $method
-     * @param  mixed  $statements
-     * @return Node|null
-     */
-    private function findMethod(string $method, $statements): ?Node
+    public function isRelationWithDefault(MethodReflection $methodReflection): bool
```

Comment:
i'd be interested to see what the performance impact of this is on a relatively large codebase that is making heavy use of relations, there are ways we can mitigate the impact but i feel like there has to be a simpler way.

for example the `BelongsTo` could have it's template type widened to `?Model` rather than just `Model` and that would allow the developer writing the relationship to be explicit: `BelongsTo<Model>` when `withDefault()` and `BelongsTo<?Model>` without it.

however this requires a breaking change to how relationships are handled right now so i fear it's too late, but i wanted to put the idea forwards regardless

Reply:
After thinking about it more, parsing the file twice is just worst case---it could parse the file twice if the conditions align just right, or once, or it might not parse it at all.

Not sure that it is worth the optimization, but we might could cache the file parse for later use---but this would increase the memory kept around